src/credentials/FabricTable.h

```diff
@@ -303,8 +261,23 @@ class ConstFabricIterator
         return other;
     }
 
-    const FabricInfo & operator*() const { return mStart[mIndex]; }
-    const FabricInfo * operator->() const { return mStart + mIndex; }
+    const FabricInfo & operator*() const
+    {
+        if (IsAtEnd())
```

Comment:
This is not a sane function to call if `IsAtEnd()`, right?  Should just `VerifyOrDie(!IsAtEnd())`.